### PR TITLE
[mask_rom] Refactor alert_handler to use sec_mmio

### PR DESCRIPTION
**Pull request by cfrantz**
There are two commits in this PR:

1. Integrate sec_mmio with shutdown: integrates the sec_mmio module with the shutdown module in the event of a fault during IO or checking expectations.
2. Update the alert_handler driver to use sec_mmio to write to the alert_handler's registers.

### CODE REVIEW

File: sw/device/silicon_creator/lib/base/sec_mmio.c [hunk 1]
```diff
@@ -78,7 +78,22 @@ void sec_mmio_write32(uint32_t addr, uint32_t value) {
...
   if ((abs_mmio_read32(addr) ^ kSecMmioMaskVal) != masked_value) {
-    sec_mmio_shutdown_cb();
+    sec_mmio_shutdown_cb(kErrorSecMmioWriteFault);
+    __builtin_unreachable();
+  }
+  ++sec_mmio_ctx.write_count;
+}
+
+void sec_mmio_write32_shadowed(uint32_t addr, uint32_t value) {
+  // Shadowed register require two writes.
```
Comment (moidx) [inline sw/device/silicon_creator/lib/base/sec_mmio.c:88]: nit: s/require/requires/

File: sw/device/silicon_creator/lib/base/sec_mmio.h [hunk 2]
```diff
@@ -151,6 +153,25 @@ uint32_t sec_mmio_read32(uint32_t addr);
  */
 void sec_mmio_write32(uint32_t addr, uint32_t value);
 
+/**
+ * Writes an aligned uint32_t to the MMIO region `base` at the give byte
+ * `offset`.
+ *
+ * This function implements a write-write-read-comparison operation for shadowed
+ * registers.. The first write value is stored in the list of expected register
```
Comment (moidx) [inline sw/device/silicon_creator/lib/base/sec_mmio.h:161]: nit: .. --> .

**Review by moidx — APPROVED** [review]

File: sw/device/silicon_creator/lib/base/sec_mmio.c (hunk 1 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/base/sec_mmio.c]: Fixed: "Shadowed registers require two writes."

File: sw/device/silicon_creator/lib/base/sec_mmio.h (hunk 2 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/base/sec_mmio.h]: Fixed.

File: sw/device/silicon_creator/lib/base/mock_sec_mmio.h [hunk 3]
```diff
@@ -50,6 +51,22 @@ using MockSecMmio = testing::StrictMock<internal::MockSecMmio>;
...
+ * Expect a shadowed write to the given offset with the given 32-bit value.
+ *
+ * The value may be given as an integer, a pointer to little-endian data,
+ * or a `std::initializer_list<BitField>`.
+ *
+ * This function is only available in tests using a fixture that derives
+ * `MmioTest`.
+ *
+ * This expectation is sequenced with all other `EXPECT_SEC_READ` and
+ * `EXPECT_SEC_WRITE` calls.
```
Comment (alphan) [inline sw/device/silicon_creator/lib/base/mock_sec_mmio.h:64]:
```suggestion
 * @param addr Write address.
 * @param ... Expected value to be written. May be an integer, a pointer to
 * little-endian data, or a `std::initializer_list<BitField>`.
```

File: sw/device/silicon_creator/lib/error.h [hunk 4]
```diff
@@ -21,21 +21,25 @@ extern "C" {
  * Choose a two-letter identifier for each module and encode the module
  * identifier the concatenated ASCII representation of those letters.
  */
+#define MODULE_CODE(ch0_, ch1_) ((ch0_) << 8 | (ch1_))
```
Comment (alphan) [inline sw/device/silicon_creator/lib/error.h:24]: note: This reverses the byte order when compared to previous scheme (just wanted to make sure that this was intended).

**Review by alphan — APPROVED** [review]
Thanks @cfrantz and apologies for the conflict due to #7516.

File: sw/device/silicon_creator/lib/base/mock_sec_mmio.h (hunk 3 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/base/mock_sec_mmio.h:64]: Fixed.

File: sw/device/silicon_creator/lib/error.h (hunk 4 above)
Comment (cfrantz) [reply sw/device/silicon_creator/lib/error.h:24]: This is intended.  The previous expression of the module codes as hex-constants was inconsistent - some where in opposite orders.

The ordering here causes the printed `boot_fault` code from `shutdown_finalize` to be in the natural human expected order.

Example: when the current mask ROM fails to find a ROM_EXT image in mask_rom.c:220:

```
RETURN_IF_ERROR(romextimage_manifest_get(kFlashSlotA, &manifest));
```

The error code is `kErrorRomextimageInternal`, which then gets printed as:

```
boot_fault: 0x0252450d
```

The middle two bytes 0x5245 are the module code. By way of asciitable.com, we see this module 'RE'.

**Pull request merged**